[JAMES-3943] Upstream cassandra removal and jib packaging

jeantil opened the pull request.

Arsnael commented on backends-common/pulsar/src/main/scala/org/apache/james/backends/pulsar/PulsarConfiguration.scala:
```diff
@@ -1,21 +1,21 @@
  * specific language governing permissions and limitations      *
  * under the License.                                           *
- ****************************************************************/
+ * ************************************************************** */
```
Some format issue on the license?

jeantil replied:
I fixed it thanks for spotting

Arsnael reviewed (commented):
Just a small typo comment but looks good to me otherwise :)

jeantil commented:
The build has been [green](https://ci-builds.apache.org/blue/organizations/jenkins/james%2FApacheJames/detail/PR-1742/9/pipeline/) at least once . 

I pushed an additional change:  I updated the app README as it mentionned that the cassandra dependency would eventually be removed. It seems we also switched from md to adoc. I don't really remember doing this :sweat:  but I guess it not really a problem.

jeantil commented:
I noticed a typo in the jib config that prevented the extension jars to be properly accounted for. I rebased on upstream and squashed the fix directly  but the changed line is https://github.com/apache/james-project/pull/1742/files#diff-c558115f4f50e3e55120ce47b62ba326f8bb5e7446d554f04c411de51db8ad5aR297 ( it was /root/extension-jars instead of /root/extension**s**/jar )
